src/uv-common.c

```diff
@@ -33,6 +33,62 @@
 # include <net/if.h> /* if_nametoindex */
 #endif
 
+
+static uv_allocator_t uv__allocator = {
+  malloc,
+  realloc,
+  free,
+};
+
+void* uv__calloc(size_t count, size_t size) {
+  /* Note: the multiplcation is not safe if
+   * the result overflows a 'size_t' */
+  size_t total = count * size;
+  void* m = uv__malloc(total);
+  if (m == NULL)
+    return NULL;
+  return memset(m, 0, total);
+}
+
+char* uv__strdup(const char *s) {
+  size_t len = strlen(s) + 1;
+  char* m = uv__malloc(len);
+  if (m == NULL)
+    return NULL;
+  return memcpy(m, s, len);
+}
+
+char* uv__strndup(const char *s, size_t n) {
+  size_t len = strnlen(s, n);
```

Comment:
I believe strnlen() requires  `-D_XOPEN_SOURCE=700` (or `-D_GNU_SOURCE` on Linux.)  It's a POSIX.1-2008 thing.